packages/typespec-azure-resource-manager/src/rules/arm-no-record.ts

```diff
@@ -0,0 +1,47 @@
+import { Model, SemanticNodeListener, createRule } from "@typespec/compiler";
+
+function isRecordType(model: Model): boolean {
+  return model.name === "Record";
+}
+
+export const armNoRecordRule = createRule({
+  name: "arm-no-record",
+  severity: "warning",
+  description: "Don't use Record types for ARM resources.",
+  url: "https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/rules/no-record",
+  messages: {
+    default:
+      "Definitions must not be of type Record. ARM requires Resource provider teams to define types explicitly.",
+    extends:
+      "Definitions must not extend type Record. ARM requires Resource provider teams to define types explicitly.",
+    is: "Definitions must not equate to type Record. ARM requires Resource provider teams to define types explicitly.",
+  },
+  create(context): SemanticNodeListener {
+    return {
+      model: (model: Model) => {
```

Comment:
Do we have a mechanism for omitting types in Azure.ResourceManager?  We want to make sure that valid usages of `Record<unknown>`, like 'tags' for example, are exempted.

Reply:
I'm still working on that bit.

Reply:
I added a test for this... and it passes... but I don't understand _why_ it passes....

Reply:
Is passes because of the design of the linter rule framework. Using `ArmTagsProperty` triggers the rule to emit a diagnostic, but because it originated outside of user code, it omits the diagnostic.